packages/legacy-client/src/client/convertWireToOsdkObject.ts

```diff
@@ -119,5 +123,131 @@ export function convertWireToOsdkObject<
   }
 
   Object.setPrototypeOf(obj, proto);
+  setPropertyAccessors<T, O>(client, apiName, obj);
+
   return obj as OsdkLegacyObjectFrom<O, T>;
 }
+
+function setPropertyAccessors<
+  T extends ObjectTypesFrom<O> & string,
+  O extends OntologyDefinition<any>,
+>(client: ThinClient<O>, apiName: T, obj: OntologyObjectV2) {
+  for (
+    const [k, v] of Object.entries(client.ontology.objects[apiName].properties)
+  ) {
+    if (!obj[k]) {
```

Comment:
You probably want to explicitly check against null here because of values like `0`